core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -300,6 +303,50 @@ public int getLostWorkerCount() {
     return workerInfoList;
   }
 
+  @Override
+  public List<ReportWorkerInfo> getReportWorkerInfoList(WorkerRange workerRange,
+      Set<ReportWorkerInfoField> fieldRange, Set<String> addresses) throws UnavailableException {
+    if (mSafeModeManager.isInSafeMode()) {
+      throw new UnavailableException(ExceptionMessage.MASTER_IN_SAFEMODE.getMessage());
+    }
+    IndexedSet<MasterWorkerInfo> selectedWorkers = null;
+    switch (workerRange) {
+      case ALL:
+        selectedWorkers = mWorkers;
+        selectedWorkers.addAll(mLostWorkers);
```

Comment:
Wow, big mistake.
I should pay attention to these kinds of mistake in the future. Thank you so much!

Changed to 
```java
      case ALL:
        selectedWorkers = new IndexedSet<>(ID_INDEX, ADDRESS_INDEX);
        selectedWorkers.addAll(mWorkers);
        selectedWorkers.addAll(mLostWorkers);
        break;
```

Reply:
to avoid issues like this, it's good to follow these rules:
- Never set a local variable equal to a field reference
  e.g. never write something like
  ```java
  selectedWorkers = mWorkers;
  ```
  This is dangerous to do because `selectedWorkers` looks like it's safe to modify since it's a local field, but in reality it is pointing to the same data structure as a field.
- Treat local variables as though they are final
  As much as possible, avoid re-assigning local variables. This makes it harder to keep track of what a local variable is actually referencing.

Reply:
For all the objects, always make a thorough copy, do not point to the same target.
So in the future, for all sets, maps, I always make a copy like
Set<..> newSet = new HashSet<>(oldSet);
Map<..> newMap = new HashMap<>(oldMap);

Thank you so much! I learn this before but never pay much attention to it. 
Some big bug may occur from this kind of negligence.